src/bridge/useBridgeTransaction.js

```diff
@@ -179,16 +179,32 @@ const useBridgeTransaction = (optionalInit?: ?() => $Shape<State>): Result => {
   const mainAccount = account ? getMainAccount(account, parentAccount) : null;
 
   const errorDelay = useRef(INITIAL_ERROR_RETRY_DELAY);
+  const statusIsPending = useRef(false); // Stores if status already being processed
+
+  const bridgePending = transaction !== statusOnTransaction;
 
   // when transaction changes, prepare the transaction
   useEffect(() => {
     let ignore = false;
     let errorTimeout;
+
+    // If bridge is not pending, transaction change is due to
+    // the last onStatus dispatch (prepareTransaction changed original transaction) and must be ignored
+    if (!bridgePending) return;
+
     if (mainAccount && transaction) {
-      Promise.resolve()
+      // We don't debounce first status refresh, but any subsequent to avoid multiple calls
+      // First call is immediate
+      const debounce = statusIsPending.current
+        ? delay(DEBOUNCE_STATUS_DELAY)
+        : null;
+
+      statusIsPending.current = true; // consider pending until status is resolved (error or success)
+
+      Promise.resolve(debounce)
```

Comment:
I'm not sure that `Promise.resolve(promise)` actually awaits the promise.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve

```The Promise.resolve() method returns a Promise object that is resolved with a given value. If the value is a promise, that promise is returned; if the value is a thenable (i.e. has a "then" method), the returned promise will "follow" that thenable, adopting its eventual state; otherwise the returned promise will be fulfilled with the value.```

It permits passing a promise or a raw value directly (here null) and cast it as a Promise in the next steps.
For what i have tested, this is working as said on Polkadot.

But i did not tested this with immediate status on other coins.
On polkadot, i have almost immediate value once fee estimation is in cache, and i tried triggering multiple updates as fast as possible but clicking fast on a checkbox:
* if fees are not cached, it estimate fees only twice (first click and last)
* if fees are cache, the status is updated quickly, but if i'm fast enough (quicker than getting a response) the status starts to debounce and only renders at the end.